public/js/action-list.js

```diff
@@ -76,74 +83,378 @@
                     }
+
+                    break;
+            }
+
+            // $currentActiveItems already contain the first/last element of the list and have no prev/next element
+            if (! toActiveItem) {
+                return;
+            }
+
+            _this.setActive(toActiveItem, markAsLastActive);
+        }
+
+        getDirectionalNext(item, eventKey) {
+            if (! item) {
+                return null;
+            }
+
+            return eventKey === 'ArrowUp' ? item.previousElementSibling : item.nextElementSibling;
+        }
+
+        findToActiveItem(lastActivatedItem, eventKey) {
+            let toActiveItem;
+            let markAsLastActive;
+
+            toActiveItem = this.getDirectionalNext(lastActivatedItem, eventKey);
+
+            while (toActiveItem) {
+                if (! toActiveItem.classList.contains('active')) {
+                    break;
                 }
 
-                _this.icinga.loader.loadUrl(
-                    url, _this.icinga.loader.getLinkTargetFor($target)
-                );
+                toActiveItem = this.getDirectionalNext(toActiveItem, eventKey);
+            }
+
+            markAsLastActive = toActiveItem;
+            // if the next/previous sibling element is already active, mark the last/first active element in list as last active
+            while (markAsLastActive && this.getDirectionalNext(markAsLastActive, eventKey)) {
+                if (! this.getDirectionalNext(markAsLastActive, eventKey).classList.contains('active')) {
+                    break;
+                }
+
+                markAsLastActive = this.getDirectionalNext(markAsLastActive, eventKey);
+            }
+
+            return [toActiveItem, markAsLastActive];
+        }
+
+        selectAll(list) {
+            let toActive = list.querySelectorAll(':scope > [data-action-item]:not(.active)');
+
+            if (toActive.length) {
+                toActive.forEach(item => item.classList.add('active'));
+                this.setActive(toActive, list.lastChild);
+            }
+        }
+
+        clearSelection(selectedItems) {
+            selectedItems.forEach(item => item.classList.remove('active'));
+        }
+
+        setActive(toActiveItem, markAsLastActive) {
+            if (toActiveItem instanceof HTMLElement) {
+                toActiveItem.classList.add('active');
+
+                toActiveItem.nextElementSibling
+                    ? toActiveItem.nextElementSibling.scrollIntoView({block: "nearest"})
+                    : null;
+                toActiveItem.previousElementSibling
+                    ? toActiveItem.previousElementSibling.scrollIntoView({block: "nearest"})
+                    : null;
+            }
+
+            this.lastActivatedItemUrl = markAsLastActive
+                ? markAsLastActive.dataset.icingaDetailFilter
+                : toActiveItem.dataset.icingaDetailFilter;
+
+            let list = document.querySelector('.action-list');
```

Comment:
There may be not just one action list. Think of an alternative way of identifying the list.